vllm/utils.py

```diff
@@ -1108,6 +1112,104 @@ def parse_args(self, args=None, namespace=None):
 
         return super().parse_args(processed_args, namespace)
 
+    @staticmethod
+    def _pull_args_from_config(args: List[str]) -> List[str]:
+        """
+            method to pull arguments specified in the config file
+            into the commandline args variable.
+            
+            The arguments in config file will be inserted between 
+            the argument list.
+            
+            example:
+            ```yaml
+                port: 12323
+                tensor-parallel-size: 4
+            ```
+            ```python
+            $: vllm {serve,chat,complete} "facebook/opt-12B" \
+                --config config.yaml -tp 2
+            $: args = [
+                "serve,chat,complete",
+                "facebook/opt-12B", 
+                '--config', 'config.yaml', 
+                '-tp', '2'
+            ]
+            $: args = [
+                "serve,chat,complete",
+                "facebook/opt-12B", 
+                '--port', '12323', 
+                '--tensor-parallel-size', '4', 
+                '-tp', '2'
+                ]
+            ```
+
+            Please note how the config args are inserted after the sub command.
+            this way the order of priorities is maintained when these are args 
+            parsed by super().
+        """
+        assert args.count(
+            '--config') <= 1, "More than one config file specified!"
+
+        index = args.index('--config')
+        if index == len(args) - 1:
+            raise ValueError("No config file specified! \
+                             Please check your commandline arguments.")
+
+        file_path = args[index + 1]
+        extension: str = file_path.split('.')[-1]
+        if extension not in {'yaml', 'yml'}:
+            raise ValueError(
+                "Config file must be of a yaml/yml type.\
+                              %s supplied", extension)
```

Comment:
Imo this part should be inside `_load_config_file`.

Reply:
done.